src/DeploymentProvider/DeploymentHelper.ts

```diff
@@ -3,22 +3,50 @@ import { AppPlatformManagementClient, AppPlatformManagementModels as Models } fr
 import { uploadFileToSasUrl } from "./azure-storage";
 import * as core from "@actions/core";
 import { parse } from 'azure-actions-utility/parameterParserUtility';
+import {deploymentName} from "@azure/arm-appplatform/esm/models/parameters";
 
 export class DeploymentHelper {
 
     private static readonly CREATE_OR_UPDATE_SUCCESS_CODE: Array<number> = [200];
     private static readonly DELETE_SUCCESS_CODE: Array<number> = [200, 204];
     private static readonly GET_SUCCESS_CODE: Array<number> = [200];
 
+    private static listDeploymentsResult: Models.DeploymentsListResponse = null;
+
     private static async listDeployments(client: AppPlatformManagementClient, params: ActionParameters): Promise<Models.DeploymentsListResponse> {
+        if (this.listDeploymentsResult != null) {
+            core.debug('list from cache, list deployments response: ' + this.listDeploymentsResult._response.bodyAsText);
+            return this.listDeploymentsResult;
+        }
         const deployments: Models.DeploymentsListResponse = await client.deployments.list(params.resourceGroupName, params.serviceName, params.appName);
         core.debug('list deployments response: ' + deployments._response.bodyAsText);
         if (!this.GET_SUCCESS_CODE.includes(deployments._response.status)) {
             throw Error('ListDeploymentsError');
         }
+        this.listDeploymentsResult = deployments;
         return deployments;
     }
 
+    private static async getDeployment(client: AppPlatformManagementClient, params: ActionParameters, deploymentName: string): Promise<Models.DeploymentResource> {
+        if (this.listDeploymentsResult != null) {
+            core.debug('get from list cache, list deployments response: ' + this.listDeploymentsResult._response.bodyAsText);
+            let ret: Models.DeploymentResource;
+            this.listDeploymentsResult.forEach(deployment => {
+                core.debug('deployment str: ' + JSON.stringify(deployment));
+                if (deployment.name == deploymentName) {
+                    ret = deployment;
```

Comment:
just return here?

Reply:
https://stackoverflow.com/questions/2641347/short-circuit-array-foreach-like-calling-break
Typescript array forEach cannot break or return and we have at most 2 items right now. So is that OK to loop the array without break?